CodingPractice/PythonAssignments/cleancode/StringUtils.py

```diff
@@ -23,3 +23,8 @@ def first_and_last_letter(string):
 
 def get_even_index_position(string):
     return ''.join(string[0::2])
+
+
+def sorted_string(string):
```

Comment:
Can you please explain what this code is doing?
What does sorted(string) return and why?

Reply:
This code is sorting a string from lowercase to uppercase letters. Just sorting on a string returns the string sorted in alphabetical order into a new list.

```python
>>> s = 'sufi'
>>> y = sorted(s)
>>> print(y)
['f', 'i', 's', 'u']
```
Here using the key=str.isupper formats the string. In sorting, False comes before True, and isupper checks if a character is uppercase or not. In this case, as the method parses the string given to it, it returns False or True depending on the case of the letter, and puts these into a new list. Therefore will put the lowercase letters - or the False values - first. 

If using str.islower, it will put the uppercase letters first, and so you would have to reverse=True the sort to get the same result. Therefore, isupper is much more efficient and takes up less code.